scripts/ci/build_emscripten.sh

```diff
@@ -34,55 +34,98 @@
 
 set -ev
 
-if test -z "$1"; then
-	BUILD_DIR="emscripten_build"
-else
-	BUILD_DIR="$1"
-fi
+function build() {
+    local build_dir=$1
+    local pre_release=$2
 
-WORKSPACE=/root/project
+    cd /root/project
 
-cd "$WORKSPACE"
+    # shellcheck disable=SC2166
+    if [[ "$CIRCLE_BRANCH" = release || -n "$CIRCLE_TAG" || -n "$FORCE_RELEASE" || "$(git tag --points-at HEAD 2>/dev/null)" == v* ]]
+    then
+        echo -n >prerelease.txt
+    else
+        # Use the given prerelease string if provided
+        if [[ -z "$pre_release" ]]; then
+           pre_release="ci"
+        fi
+        # Use last commit date rather than build date to avoid ending up with builds for
+        # different platforms having different version strings (and therefore producing different bytecode)
+        # if the CI is triggered just before midnight.
+        TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="${pre_release}.%cd" >prerelease.txt
```

Comment:
Since the variable is no longer the whole prerelease string, but just a part of it, I'd rename it accordingly:
```suggestion
        TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="${prerelease_source}.%cd" >prerelease.txt
```